### ISSUE TITLE
[train-141][stage] packaged code is not pointed at the CDN

### ISSUE THREAD

**Issue by jrgm** [mozilla/fxa#1729]
With v1.141.1 in stage, there are dozens of errors in the server tests like:

```
#https get https://accounts.stage.mozaws.net/authorization (18s)
[19:15:06]
AssertionError: https://accounts.stage.mozaws.net/bundle-67f46f72f9c77b5c06ebadc05f347ea498b9baec/head.bundle.js should not have CORS headers: expected true to be falsy
  at <tests/server/helpers/routesHelpers.js:205:18>
```

I think the packaging-for-CDN isn't right in 1.141. If I put back v1.140.3 in stage, for the top-level '/', I see, for one example, this difference:
```
v1.140.3  <script src="https://accounts-static-2.stage.mozaws.net/bundle-59c9857e777d19e6884edac7be1b399693fb21b9/head.bundle.js" ...

v1.141.1  <script src=/bundle-67f46f72f9c77b5c06ebadc05f347ea498b9baec/head.bundle.js ...
```
So, in v1.141.1, sub-resources are coming from the origin, not the CDN.

By the way, when I put back v1.140.3, I used the same version of cloudops-deployment, so I don't think the deployment pipeline is the root cause (unless it's some new config that has been added in 141).

**Comment by jrgm** [comment]
This should be controlled by STATIC_RESOURCE_URL. I checked the docker container process running environment (`sudo cat /proc/4107/environ | xargs -0 -n 1 | grep -i static`), that it has this value in stage: `STATIC_RESOURCE_URL=https://accounts-static-2.stage.mozaws.net`.

I also pushed a branch of master to dump the config at startup, and the config object has this value: `\"static_resource_url\": \"https://accounts-static-2.stage.mozaws.net\"`,

So, not clear to me what has changed.

**Comment by philbooth** [comment]
> So, not clear to me what has changed.

Indeed. I tried diffing the respective branches for clues, although it's complicated because this is also the release where prettier landed so there's a lot of noise from formatting changes. Too much noise in some places, unfortunately.

But `Dockerfile-build` is identical:

```diff
~/c/fxa (train-141) $ git diff train-140..train-141 -- packages/fxa-content-server/Dockerfile-build
```

Nothing significant in the top-level `_scripts` dir, which could affect the content server build:

```diff
~/c/fxa (train-141) $ git diff train-140..train-141 -- _scripts
diff --git a/_scripts/install_all.sh b/_scripts/install_all.sh
index d729f0ac4..748fe2c3e 100755
--- a/_scripts/install_all.sh
+++ b/_scripts/install_all.sh
@@ -30,7 +30,7 @@ cd ..

 cd browserid-verifier; npm ci; cd ..

-cd fxa-js-client; npm ci; cd ..
+cd fxa-js-client; npm ci; npx grunt sjcl; cd ..

 cd fxa-customs-server; npm ci; cd ..

```

Ditto the `.circleci` dir:

```diff
~/c/fxa (train-141) $ git diff train-140..train-141 -- .circleci/
diff --git a/.circleci/build.sh b/.circleci/build.sh
index 9af6c44e6..1f699c4ef 100755
--- a/.circleci/build.sh
+++ b/.circleci/build.sh
@@ -20,6 +20,10 @@ if grep -e "$MODULE" -e 'all' $DIR/../packages/test.list; then
   if [ "${MODULE}" == 'fxa-oauth-server' ]; then
     cp $DIR/../packages/version.json fxa-oauth-server/config
     docker build -f Dockerfile-oauth-build -t ${MODULE}:build .
+  elif [ "${MODULE}" == 'fxa-payments-server' ]; then
+    cd ..
+    docker build -f fxa-payments-server/Dockerfile -t ${MODULE}:build .
+    cd fxa-payments-server
   elif [[ -e Dockerfile ]]; then
     docker build -f Dockerfile -t ${MODULE}:build .
     # docker run --rm -it ${MODULE}:build npm ls --production
diff --git a/.circleci/config.yml b/.circleci/config.yml
index ad752c60c..5510c0a07 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -232,6 +232,8 @@ jobs:
           rustup component add rustfmt-preview
           cargo fmt -- --check
           ./scripts/test_with_authdb.sh
+          cargo install cargo-audit
+          cargo audit
       - store_artifacts:
           path: fxa-auth-db-mysql.log
       - setup_remote_docker
diff --git a/.circleci/test-content-server.sh b/.circleci/test-content-server.sh
index ca2395477..02fd6b9d5 100755
--- a/.circleci/test-content-server.sh
+++ b/.circleci/test-content-server.sh
@@ -57,7 +57,7 @@ if grep -e "$MODULE" -e 'all' $DIR/../packages/test.list; then
     test_suite pairing

     mozinstall firefox.tar.bz2
-    # test_suite server
+    test_suite server

   else
     mozinstall firefox.tar.bz2
```

The diff for the content server itself is pretty long though.

**Comment by philbooth** [comment]
I don't understand how it could be related, but mentioning out of hyper-caution.

As you can see from the above diff, in train 141 we started building the payments server and it does the build at the `packages/` level of the tree rather than in `packages/fxa-payments-server/`.

The `Dockerfile` for the payments server forcibly sets `PUBLIC_URL` to `/`, which was a fix from #1636 to generate correct `href` attributes in the build. Is it possible that environment variable is affecting the content server build as well somehow?

It seems unlikely, but I'm searching for possible causes and this is the only thing I've come up with so far. :confused:
